timesketch/api/v1/resources.py

```diff
@@ -1658,6 +1660,85 @@ def post(self):
             status_code=HTTP_STATUS_CODE_BAD_REQUEST)
 
 
+class AnalyzerRunResource(ResourceMixin, Resource):
+    """Resource to get all timelines for sketch."""
+
+    @login_required
+    def get(self, sketch_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            A list of all available analyzer names.
+        """
+        _ = sketch_id
+        analyzers = [
+            x for x, y  in analyzer_manager.AnalysisManager.get_analyzers()]
+
+        return analyzers
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to the resource.
+
+        Returns:
+            A string with the response from running the analyzer.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        form = RunAnalyzerForm.build(request)
+
+        if not form.validate_on_submit():
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST, 'Unable to validate input data.')
+
+        if not sketch.has_permission(current_user, 'write'):
+            return abort(
+                HTTP_STATUS_CODE_FORBIDDEN,
+                'User does not have write permission on the sketch.')
+
+        timeline_name = form.timeline_name.data
+        timeline_id = form.timeline_id.data
+
+        search_index = None
+        if timeline_name:
+            for timeline in sketch.timelines:
```

Comment:
Hmm, i see what you are doing here. You will analyze all timelines with the name. Is this what the caller expects?

Reply:
changed